yt/geometry/coordinates/tests/test_cartesian_coordinates.py

```diff
@@ -13,19 +13,39 @@ def test_cartesian_coordinates():
     # We're going to load up a simple AMR grid and check its volume
     # calculations and path length calculations.
     ds = fake_amr_ds()
+    print("ds initialized", flush=True)
     axes = sorted(set(ds.coordinates.axis_name.values()))
-    for i, axis in enumerate(axes):
-        dd = ds.all_data()
+    print("axes set", flush=True)
+
+    dd = ds.all_data()
+    print("dd loaded", flush=True)
+
+    i = 0
+    for axis in axes:
```

Comment:
I don't like this change.  I'm surprised it is necessary for Python 3.12.

Reply:
Like most changes currently bundled in this draft PR, it's not. This is a temporary commit, and a desperate attempt to debug this test. At the moment, I have *no* idea why it crashes, and I haven't been able to reproduce the problem locally.

Reply:
:+1: